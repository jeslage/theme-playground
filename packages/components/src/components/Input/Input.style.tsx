import styled from 'styled-components';

const StyledInput = styled.label`
  display: flex;
  align-items: center;
  width: 100%;

  input {
    width: 100%;
    border: none;
    appeareance: none;
    resize: none;
    background: none;
    padding: 10px;
    border-radius: 5px;
    text-align: right;
    max-width: 200px;
    border: 1px solid ${({ theme }) => theme.text};
    color: ${({ theme }) => theme.text};
    outline: none;
  }
`;

export default StyledInput;
