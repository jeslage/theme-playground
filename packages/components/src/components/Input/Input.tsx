import * as React from 'react';

import StyledInput from './Input.style';
import Label from '../Label/Label';

export interface Props {
  label: string;
  description?: string;
  iconBefore?: JSX.Element;
  onChange: (val: string) => void;
  value?: string;
}

const Input: React.FC<Props> = ({
  label,
  iconBefore,
  description,
  value,
  onChange
}) => {
  return (
    <StyledInput>
      <Label iconBefore={iconBefore} label={label} description={description} />

      <input
        type="text"
        value={value}
        onChange={event => {
          const { value: eventValue } = event.target;
          onChange(eventValue);
        }}
      />
    </StyledInput>
  );
};

export default Input;
