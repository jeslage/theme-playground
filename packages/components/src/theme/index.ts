export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
