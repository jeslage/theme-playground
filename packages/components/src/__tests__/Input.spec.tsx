import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';

import Input from '../components/Input/Input';
import StyledInput from '../components/Input/Input.style';
import { darkTheme } from '../theme';

describe('Input', () => {
  it('Input should render correctly', () => {
    const component = shallow(
      <Input
        onChange={value => console.log(value)}
        label="Label"
        description="Testbeschreibung"
        value="Testinput"
        iconBefore={
          <svg height="24" viewBox="0 0 24 24" width="24">
            <rect height="2" rx="1" width="12" x="6" y="11" />
          </svg>
        }
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('StyledInput should render correctly', () => {
    const component = shallow(<StyledInput theme={{ colors: darkTheme }} />);

    expect(component).toMatchSnapshot();
  });
});
