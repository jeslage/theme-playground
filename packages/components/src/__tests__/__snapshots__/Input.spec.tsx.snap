// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input should render correctly 1`] = `
<styled.label>
  <Label
    description="Testbeschreibung"
    iconBefore={
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <rect
          height="2"
          rx="1"
          width="12"
          x="6"
          y="11"
        />
      </svg>
    }
    label="Label"
  />
  <input
    onChange={[Function]}
    type="text"
    value="Testinput"
  />
</styled.label>
`;

exports[`Input StyledInput should render correctly 1`] = `
<label
  className="sc-AxjAm iaCQVm"
/>
`;
